### PR TITLE
[BEAM-1949] Rename SideOutputValue to OutputValue

### PR DESCRIPTION
**Pull request by tgroh**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by tgroh** [comment]
R: @aaltay

**Comment by aaltay** [comment]
Run Python PostCommit

**Comment by aaltay** [comment]
Thank you @tgroh. This LGTM. (I can merge it after the tests.)

**Comment by aaltay** [comment]
There is a lint error, could you fix that?

**Comment by tgroh** [comment]
Yes. Also I've found a few more places where "side output" is used, so I'm fixing those as well

**Comment by tgroh** [comment]
Updated. PTAL

File: sdks/python/apache_beam/examples/snippets/snippets_test.py [hunk 1]
```diff
@@ -176,55 +176,55 @@ def process(self, element, lower_bound, upper_bound=float('inf')):
     # [END model_pardo_side_input_dofn]
     self.assertEqual({'a', 'bb', 'ccc'}, set(small_words))
 
-  def test_pardo_with_side_outputs(self):
-    # [START model_pardo_emitting_values_on_side_outputs]
+  def test_pardo_with_tagged_outputs(self):
+    # [START model_pardo_emitting_values_on_tagged_outputs]
```
Comment (aaltay) [inline sdks/python/apache_beam/examples/snippets/snippets_test.py]: web site needs to be updated for these changes in snippet names. Please sync with: @hadarhod87 @melap

Comment (melap) [inline sdks/python/apache_beam/examples/snippets/snippets_test.py]: it should pretty straightforward, though some Python code comments that aren't part of the snippets will also need to be updated in addition to the snippet tag names. I can create a PR to update these in the doc files if you want @tgroh , let me know

**Comment by tgroh** [comment]
Run Python PostCommit

**Comment by aaltay** [comment]
LGTM

**Comment by robertwb** [comment]
Thanks for doing this. In terms of naming, all values are OutputValue, even the main one--could we call this TaggedOutput instead. Thus one would write, e.g.

```
def process(element):
    yield element + 1 # main
    yield TaggedOutput(element + 2) # non-main
```

**Pull request closed without merging**